feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts

```diff
@@ -138,12 +139,40 @@ export class ConfiguratorStorefrontUtilsService {
     }
   }
 
+  protected getFocusableElementById(
+    focusableElements: HTMLElement[],
+    id?: string
+  ): HTMLElement | undefined {
+    return focusableElements.find((focusableElement) => {
+      if (id) {
+        if (focusableElement.id.indexOf(id) !== -1) {
+          return focusableElement;
+        }
+      }
+    });
+  }
+
+  protected getFocusableElementByValueUiKey(
+    focusableElements: HTMLElement[],
+    valueUiKey?: string
+  ): HTMLElement | undefined {
+    return this.getFocusableElementById(focusableElements, valueUiKey);
+  }
+
+  protected getFocusableElementByAttributeId(
+    focusableElements: HTMLElement[],
+    attributeName: string
+  ): HTMLElement | undefined {
+    return this.getFocusableElementById(focusableElements, attributeName);
+  }
+
   /**
-   * Find an attribute by its name in the form and focus it.
+   * Find a value by its UI key in the form and focus it.
    *
-   * @param {string} name - Attribute name
+   * @param {string} attributeName - Attribute name
+   * @param {string} valueUiKey - Value UI key
    */
-  focusAttribute(name: string): void {
```

Comment:
removing a method is an incompatible change - are you sure about this?

Reply:
This method occurred in the period as the conflict solver navigation had been implemented that's why no incompatible change is expected